Requests refactor RFC

lucaslcode opened the pull request:
See comments.

Also renamed GroupChats to Chats

lucaslcode committed:
request refactor proto

lucaslcode committed:
fix

lucaslcode commented on app/proto/conversations.proto:
```diff
@@ -11,63 +11,59 @@ import "annotations.proto";
     // Raises: NOT_FOUND if there is no such direct message thread
   }
 
-  rpc GetUpdates(GetUpdatesReq) returns (GetUpdatesRes) {
-    // Returns all new messages (aka updates) since a given message id
```
This wasn't being used since we switched to react-query

lucaslcode commented on app/proto/conversations.proto:
```diff
@@ -102,17 +98,25 @@ message MessageContentUserRemoved {
 }
 
 message MessageContentHostRequestStatusChanged {
+  uint64 host_request_id = 1;
```
Since a thread may have multiple host requests in it

lucaslcode committed:
put back last_seen_message_id [skip-ci]

lucaslcode committed:
fix group_chat, add chat_id to request [skip-ci]

lucaslcode committed:
change uint64 ids to int64 [skip-ci]

lucaslcode reviewed (commented):
I did not attempt to number message fields for backwards compatibility since I don't think we are planning to make this backwards compatible with old clients.

ShmuelTreiger commented on app/proto/conversations.proto:
```diff
@@ -108,16 +110,18 @@ enum HostRequestStatus {
   HOST_REQUEST_STATUS_REJECTED = 2;
   HOST_REQUEST_STATUS_CONFIRMED = 3;
   HOST_REQUEST_STATUS_CANCELLED = 4;
+  HOST_REQUEST_STATUS_MAYBE_ACCEPTED = 5;
```
What is maybe accepted?

lucaslcode replied:
If the host says "maybe" - I thought HOST_REQUEST_STATUS_MAYBE isn't clear if the host or surfer is saying maybe

ShmuelTreiger replied:
I didn't know hosts could say maybe. Seems weird to me. Was this a requested feature? In any event, code looks good.

lucaslcode replied:
Yes, both from design and people on the forum

ShmuelTreiger commented on app/proto/conversations.proto:
```diff
@@ -84,22 +80,28 @@ message MessageContentChatCreated {}
 
 message MessageContentUserInvited {
-  uint64 target_user_id = 1;
+  int64 target_user_id = 1;
```
Can these ever be negative? Why the switch to int?

lucaslcode replied:
More recent proto files like communities use int64 so I changed it to match

lucaslcode replied:
Also, I think the reason was js doesn't support uints so int64 is closer?

ShmuelTreiger commented on app/proto/requests.proto:
```diff
@@ -90,47 +78,5 @@ message ListHostRequestsReq {
-
-message GetHostRequestMessagesReq {
-  int64 host_request_id = 1;
-  int64 last_message_id = 2;
-  uint32 number = 3;
-}
-
-message GetHostRequestMessagesRes {
-  repeated org.couchers.api.conversations.Message messages = 1;
-  int64 last_message_id = 4;
-  bool no_more = 5;
-}
-
-message SendHostRequestMessageReq {
-  int64 host_request_id = 1;
-  string text = 2;
-}
-
-message GetHostRequestUpdatesReq {
-  int64 newest_message_id = 1;
-  uint32 number = 2;
-  bool only_sent = 3;
-  bool only_received = 4;
-}
-
-message HostRequestUpdate {
-  int64 host_request_id = 1;
-  org.couchers.api.conversations.Message message = 2;
-  // the current status of the host request, regardless of this update
-  org.couchers.api.conversations.HostRequestStatus status = 3;
-}
-
-message GetHostRequestUpdatesRes {
-  repeated HostRequestUpdate updates = 1;
-  bool no_more = 2;
-}
-
-message MarkLastSeenHostRequestReq {
-  int64 host_request_id = 1;
-  int64 last_seen_message_id = 2;
```
I assume this was all deleted because the front end doesn't use it? Or was this just wrapped up into the general message stuff? In either event, do we need to remove backend functions and tests?

ShmuelTreiger reviewed (commented):
Why rename from group chat? Or I guess, why did we have them as group chat in the first place? I seem to recall a whole long discussion about it, but it's been a while.

lucaslcode commented:
Rename because I don't know why they were called group chats in the first place. Right now, DMs are group chats with only 2 participants.

kalvdans commented:
I like the unification and also the rename from groupchat to chat :+1:

lucaslcode committed:
change to target_user_id
this matches the other messages and the target for a host request is obviously the host

lucaslcode committed:
preserve backwards compatibility
otherwise it messes up logging/tracing, apparently

aapeliv commented on app/proto/conversations.proto:
```diff
@@ -108,16 +109,18 @@ enum HostRequestStatus {
+  int64 host_request_id = 2;
   // this is the new host request status
-  HostRequestStatus status = 1;
+  HostRequestStatus status = 3;
```
This one is fine to keep as `1`

lucaslcode committed:
Update conversations.proto
